File: python/reverse_linked_list.py
def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        dummy = ListNode(0)
        
        while head != None :
            tmp = head.next
            head.next = dummy.next
            dummy.next = head
            head = tmp
        return dummy.next

File: java/Reverse Linked List.java
public ListNode reverseList(ListNode head) {
        
        ListNode dummy = new ListNode(0);
        
        while (head != null) {
            
            ListNode temp = head.next;
            
            head.next = dummy.next;
            
            dummy.next = head;
            
            head = temp;
            
        }
        
        return dummy.next;   
        
    }
